jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1158,13 +1196,21 @@ public Method getDeclaredMethod(String name, Class<?>... parameterTypes) throws
 /**
  * Answers the class which declared the class represented
  * by the receiver. This will return null if the receiver
- * is a member of another class.
+ * is not a member of another class.
  *
  * @return		the declaring class of the receiver.
  */
 @CallerSensitive
 public Class<?> getDeclaringClass() {
-	Class<?> declaringClass = getDeclaringClassImpl();
+	if (cachedDeclaringClassOffset == -1) {
+		cachedDeclaringClassOffset = getFieldOffset("cachedDeclaringClass"); //$NON-NLS-1$
+	}
+	if (cachedDeclaringClass == null) {
+		Class<?> localDeclaringClass = getDeclaringClassImpl();
+		writeFieldValue(cachedDeclaringClassOffset, localDeclaringClass != null ? localDeclaringClass: Dummy.class);
+	}
+
+	Class<?> declaringClass = cachedDeclaringClass != Dummy.class ? cachedDeclaringClass : null;
```

Comment:
A comment is needed to explain why we do this kind of variable initialization. 
In addition, pls re-run benchmark tests against this PR.